mcs/tools/linker/Makefile

```diff
@@ -2,21 +2,40 @@ thisdir = tools/linker
 SUBDIRS =
 include ../../build/rules.make
 
-RESOURCES = \
-	Descriptors/mscorlib.xml	\
-	Descriptors/System.xml		\
-	Descriptors/System.Core.xml		\
-	Descriptors/System.Drawing.xml	\
-	Descriptors/System.Web.xml		\
-	Descriptors/Mono.Posix.xml		\
-
 PROGRAM = monolinker.exe
 
-$(PROGRAM): $(RESOURCES)
-
 LIB_REFS = System System.Core System.Xml Mono.Cecil
-LOCAL_MCS_FLAGS = $(RESOURCES:%=-resource:%)
 
-EXTRA_DISTFILES = $(RESOURCES)
+TEST_CASES = \
+	mscorlib/test-array.cs
+
+ifndef AOT_FRIENDLY_PROFILE
+TEST_CASES += \
+	mscorlib/test-reflection.cs
+endif
+
+TESTS_COMPILER = $(MCS) -nologo -noconfig -debug:portable -r:$(topdir)/class/lib/$(PROFILE)/mscorlib.dll
+
+check: compile-tests run-tests
+
+compile-tests: $(TEST_CASES)
+
+mscorlib/test-%.cs:
```

Comment:
A makefile target like that should always produce an output matching it's name (otherwise it's a .PHONY target), that's what make's dependency logic is built around.

This produces e.g. Tests/test-array.exe, not mscorlib/test-array.cs. I'll show you how I'd do it in a bit, much easier to talk about a diff :)